cdap-api/src/main/java/io/cdap/cdap/api/plugin/PluginConfigurer.java

```diff
@@ -95,4 +101,33 @@
   @Nullable
   <T> Class<T> usePluginClass(String pluginType, String pluginName,
                               String pluginId, PluginProperties properties, PluginSelector selector);
+
+  /**
+   * Evaluates lookup macros and the 'secure' macro function using provided macro evaluator.
+   *
+   * @param properties key-value map of properties to evaluate
+   * @param evaluator  macro evaluator to be used to evaluate macros
+   * @return map of evaluated macros
+   * @throws InvalidMacroException indicates that there is an invalid macro
+   */
+  default Map<String, String> evaluateMacros(Map<String, String> properties,
+                                             MacroEvaluator evaluator) throws InvalidMacroException {
+    return evaluateMacros(properties, evaluator,
+                          MacroParserOptions.builder()
```

Comment:
MacroParserOptions.DEFAULT

The secure macro is evaluated by the app level and is not a special thing in the platform.